src/Core/Content/Product/SalesChannel/Detail/ProductDetailRoute.php

```diff
@@ -120,18 +123,31 @@ private function addFilters(SalesChannelContext $context, Criteria $criteria): v
      */
     private function checkVariantListingConfig(string $productId, SalesChannelContext $context): ?string
     {
-        /** @var SalesChannelProductEntity|null $product */
-        $product = $this->productRepository->search(new Criteria([$productId]), $context)->first();
```

Comment:
Can you try to use here partial data loading aka

$criteria->addFields(['parentId', 'variantListingConfig')):

🤔